### ISSUE TITLE
Blog plugin fails with StopIteration when no nav entry is present

### ISSUE THREAD

**Issue by paulmelis** [squidfunk/mkdocs-material#6041]
### Context

As requested in #6039 here's a reproduction of the issue described there.

### Bug description

I'm using the blog plugin to provide a "News" section on an mkdocs-material site. This includes a name different than "blog" for the `blog_dir` setting. When there is no entry for the `news/index.md` file the serving/building of the site fails with a `StopIteration` in the blog plugin:

```
  File "/home/melis/courses/blender-course/env/lib/python3.11/site-packages/material/plugins/blog/plugin.py", line 690, in _attach_to
    tail = next(item for item in reversed(items) if isinstance(item, Page))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration
```

### Related links

- https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/

### Reproduction

[9.3.2-blog-build-fails.zip](https://github.com/squidfunk/mkdocs-material/files/12657326/9.3.2-blog-build-fails.zip)

### Steps to reproduce

1. Unzip reproduction and set up virtualenv with given requirements.lock.txt
2. `mkdocs serve`
3. Exception trace:

```
(env) melis@juggle 10:10:/tmp/doh21/9.3.2-blog-build-fails$ mkdocs serve
INFO    -  Building documentation...
INFO    -  Cleaning site directory
WARNING -  A relative path to 'index.md' is included in the 'nav' configuration, which is not found in the documentation files.
Traceback (most recent call last):
  File "/tmp/doh21/9.3.2-blog-build-fails/env/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/mkdocs/__main__.py", line 270, in serve_command
    serve.serve(**kwargs)
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/mkdocs/commands/serve.py", line 86, in serve
    builder(config)
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/mkdocs/commands/serve.py", line 67, in builder
    build(config, live_server=None if is_clean else server, dirty=is_dirty)
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/mkdocs/commands/build.py", line 311, in build
    nav = config.plugins.on_nav(nav, config=config, files=files)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/mkdocs/plugins.py", line 536, in on_nav
    return self.run_event('nav', nav, config=config, files=files)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/mkdocs/plugins.py", line 507, in run_event
    result = method(item, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/material/plugins/blog/plugin.py", line 207, in on_nav
    self._attach_to(self.blog, Section(title, views), nav)
  File "/tmp/doh21/9.3.2-blog-build-fails/env/lib/python3.11/site-packages/material/plugins/blog/plugin.py", line 688, in _attach_to
    tail = next(item for item in reversed(items) if isinstance(item, Page))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration
```

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by alexvoss** [comment]
Hi, thanks for the reproduction. I will try to look at this but am traveling across timezones today, so not sure exactly how much time I will have before Monday. Since the blog plugin is one of the newer ones, I am keen to gain an understanding of how it works and this seems like a good test case. Please bear with me, I am new to trying to fix things in mkdocs-material.

**Comment by squidfunk** [comment]
@alexvoss thanks for offering to take this on. I found some time to tackle this and I'm going to look into this right now. I'm happy to give you an introduction on the blog plugin later, as there are still some edge cases to fix ☺️

**Comment by squidfunk** [comment]
Released as part of [9.4.2](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.2).
